typing/ctype.ml

```diff
@@ -618,6 +620,8 @@ let rec generalize_structure ty =
       match get_desc ty with
         Tconstr (p, _, abbrev) ->
           not (is_object_type p) && (abbrev := Mnil; true)
+      | Tvariant row when not (is_fixed row) && not (static_row row) ->
```

Comment:
I think `row_fixed row` rather than `is_fixed row` would be fine and allow more cases to type-check.